sysbuild/CMakeLists.txt

```diff
@@ -52,38 +94,144 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
+
+    if(SB_CONFIG_MCUBOOT_MODE_SINGLE_APP)
+      set(mcuboot_mode CONFIG_SINGLE_APPLICATION_SLOT)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_SINGLE_APP)
+    elseif(SB_CONFIG_MCUBOOT_MODE_SWAP_WITHOUT_SCRATCH)
+      set(mcuboot_mode CONFIG_BOOT_SWAP_USING_MOVE)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_SWAP_WITHOUT_SCRATCH)
+    elseif(SB_CONFIG_MCUBOOT_MODE_SWAP_SCRATCH)
+      set(mcuboot_mode CONFIG_BOOT_SWAP_USING_SCRATCH)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_SWAP_SCRATCH)
+    elseif(SB_CONFIG_MCUBOOT_MODE_OVERWRITE_ONLY)
+      set(mcuboot_mode CONFIG_BOOT_UPGRADE_ONLY)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_OVERWRITE_ONLY)
+    elseif(SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP)
+      set(mcuboot_mode CONFIG_BOOT_DIRECT_XIP)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP)
+    elseif(SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP_WITH_REVERT)
+      set(mcuboot_mode CONFIG_BOOT_DIRECT_XIP;CONFIG_BOOT_DIRECT_XIP_REVERT)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT)
+    endif()
+
+    # Apply configuration to MCUboot
+    foreach(mode ${mcuboot_modes})
+      if("${mode}" IN_LIST mcuboot_mode)
+        set_config_bool(mcuboot ${mode} y)
+      else()
+        set_config_bool(mcuboot ${mode} n)
+      endif()
+    endforeach()
+
+    # Apply configuration to application
+    foreach(mode ${application_mcuboot_modes})
+      if("${mode}" IN_LIST application_mcuboot_mode)
+        set_config_bool(${DEFAULT_IMAGE} ${mode} y)
+      else()
+        set_config_bool(${DEFAULT_IMAGE} ${mode} n)
+      endif()
+    endforeach()
+
+    # Check if the board is a non-secure one
+    string(LENGTH "${BOARD}" board_len)
+    math(EXPR board_len "${board_len} - 3")
+    string(SUBSTRING "${BOARD}" ${board_len} 3 ns_board_check)
+
+    if("${ns_board_check}" STREQUAL "_ns")
+      # Configure MCUboot before application so that TF-M can read MCUboot configuration
+      sysbuild_add_dependencies(CONFIGURE ${DEFAULT_IMAGE} mcuboot)
+
+      # Configure MCUBoot to be able to boot TFM
+      add_overlay_config(
+        mcuboot
+        ${ZEPHYR_NRF_MODULE_DIR}/modules/mcuboot/tfm.conf
+        )
+
+      # Add fault injection protection to MCUBOOT
+      add_overlay_config(
+        mcuboot
+        ${ZEPHYR_NRF_MODULE_DIR}/modules/mcuboot/fih_low_enable.conf
+      )
+    endif()
+
+    # Use NCS signing script with support for PM
+    set(${DEFAULT_IMAGE}_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
+
+# hack to work around issue introduced by other hack
+    set(${DEFAULT_IMAGE}_ALT_SIGNED_HEX_FILE "${CMAKE_BINARY_DIR}/merged.hex" CACHE INTERNAL "nope" FORCE)
+
+    set(imgtool_extra)
```

Comment:
Why do we have an empty declaration?

Reply:
Clears the variable as it gets set if various conditions are present